### PR TITLE
MINOR: Fix ConnectWorkerIntegrationTest::testBrokerCoordinator

### PR DESCRIPTION
**Pull request by C0urante**
This test case has become flaky since https://github.com/apache/kafka/pull/16477 was merged.

For a quick fix, we can give workers time to discover that the Kafka cluster has gone down in order for the health check endpoint to be accurate.

Ideally, this would not be necessary and we'd always detect that the Kafka cluster is down at the top of our tick loop [here](https://github.com/apache/kafka/blob/0ada8fac6869cad8ac33a79032cf5d57bfa2a3ea/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L419). However, with some rare-but-not-impossible timing, that method may return without sending any requests to the group coordinator. This is itself probably indicative of a bug, and if reviewers agree, we can file a Jira ticket for it. But, because this possibly-buggy behavior is not newly-introduced and the cause of failure for our tests is simply a failure to account for that existing behavior in newly-introduced logic, we should not block on a fix for that behavior before modifying this test case to prevent flakiness.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by C0urante** [comment]
@gharris1727 would you mind taking a look at this? The fix is pretty straightforward, but the behavior that merits it is a little tricky; let me know if you'd like a more in-depth explanation of what I believe is causing the failures we've started seeing in CI.

**Review by gharris1727 — APPROVED** [review]
> [`ensureActive`] may return without sending any requests to the group coordinator. This is itself probably indicative of a bug, and if reviewers agree, we can file a Jira ticket for it.

This sounds to me like a bug. When I was reviewing the healthcheck endpoint, I wanted to verify that we needed to be in the group prior to returning a successful healthcheck. I saw the ensureActive and assumed that it did what it said on the tin: ensured that we were active in the group. Please do file a follow-up.

> we should not block on a fix for that behavior before modifying this test case to prevent flakiness.

I completely agree.

**Comment by C0urante** [comment]
Thanks Greg! Filed [KAFKA-17130](https://issues.apache.org/jira/browse/KAFKA-17130) to track the bug.

Will wait for CI to complete before merging.

**Pull request merged**
